idaes/models/unit_models/packed_bed_reactor.py

```diff
@@ -0,0 +1,337 @@
+        self.control_volume = ControlVolume1DBlock(default={
+                "dynamic": self.config.dynamic,
+                "has_holdup": self.config.has_holdup,
+                "property_package": self.config.property_package,
+                "property_package_args": self.config.property_package_args,
+                "reaction_package": self.config.reaction_package,
+                "reaction_package_args": self.config.reaction_package_args,
+                "transformation_method": self.config.transformation_method,
+                "transformation_scheme": self.config.transformation_scheme,
+                "finite_elements": self.config.finite_elements,
+                "collocation_points": self.config.collocation_points})
+
+        self.control_volume.add_geometry(
+                length_domain_set=self.config.length_domain_set)
+
+        self.control_volume.add_state_blocks(
+                has_phase_equilibrium=self.config.has_phase_equilibrium)
+
+        self.control_volume.add_reaction_blocks(
+                has_equilibrium=self.config.has_equilibrium_reactions)
+
+        self.control_volume.add_material_balances(
+            balance_type=self.config.material_balance_type,
+            has_rate_reactions=True,
+            has_equilibrium_reactions=self.config.has_equilibrium_reactions,
+            has_phase_equilibrium=self.config.has_phase_equilibrium)
+
+        self.control_volume.add_energy_balances(
+            balance_type=self.config.energy_balance_type,
+            has_heat_of_reaction=self.config.has_heat_of_reaction,
+            has_heat_transfer=self.config.has_heat_transfer)
+
+        self.control_volume.add_momentum_balances(
+            balance_type=self.config.momentum_balance_type,
+            has_pressure_change=self.config.has_pressure_change)
+
+        self.control_volume.apply_transformation()
+
+        # Add Ports
+        self.add_inlet_port()
+        self.add_outlet_port()
+
+        # Set references to balance terms at unit level
+        add_object_reference(self, "length", self.control_volume.length)
+
+        # Add volume variable for full reactor
+        units = self.config.property_package.get_metadata()
+        self.volume = Var(initialize=1,
+                          doc="Reactor Volume",
+                          units=units.get_derived_units("volume"))
+
+        self.area = Var(initialize=1,
+                        doc="area",
+                        units=units.get_derived_units("area"))
+
+        self.geometry = Constraint(expr=self.volume == self.config.mass_catalyst/self.config.bulk_density)
+
+        self.area_from_volume = Constraint(expr=self.area == self.volume/self.length)
+
+        def calculate_void_fraction(self):
+            return 1 - self.config.bulk_density/self.config.catalyst_density
+        self.epsilon = Expression(rule=calculate_void_fraction)
+
+        self.real_area = Constraint(expr=self.control_volume.area == self.area*
```

Comment:
I would suggest a different name for this, or at least append `_eq` to the end so that users do not mistake this for a variable giving the "real area".